src/FileIndexer.ts

```diff
@@ -65,7 +65,21 @@ export class FileIndexer {
     if (ts.isIdentifier(node) || ts.isStringLiteralLike(node)) {
       const sym = this.getTSSymbolAtLocation(node)
       if (sym) {
-        this.visitSymbolOccurrence(node, sym)
+        /**
+         * For example:
+         *
+         * const a = 1
+         *       ^ node
+         *       ^ node.parent.name
+         *       ^^^^^ node.parent
+         */
+        const isDefinition = this.declarationName(node.parent) === node
+
+        if (isDefinition) {
+          return this.indexDefinition(node, sym)
+        }
+
+        this.indexReference(node, sym)
```

Comment:
nit: Maybe `visitDefinition` and `visitReference`? We're using "index" as a verb for documents/projects, but for an AST traversal, "visit" seems more natural.